localstack/services/dynamodb/provider.py

```diff
@@ -491,8 +501,18 @@ def update_table(
             is_no_update_error = (
                 e.code == "ValidationException" and "Nothing to update" in e.message
             )
-            if is_no_update_error and update_table_input.get("ReplicaUpdates"):
-                table_name = update_table_input.get("TableName")
+            if is_no_update_error and not list(
```

Comment:
Should this condition be:
```
if not is_no_update_error or not list(...):
    raise
...
```
?

(Nice refactoring btw - this _Return Early_ pattern can help reduce the level of nesting and make the code a bit easier to read. 👍 )

Reply:
Since `is_no_update_error` should always be `true` (being in the `except` clause), aren't the two conditions equivalent? 🤔